clearwater-infrastructure/usr/share/clearwater/bin/check_config_values.py

```diff
@@ -0,0 +1,262 @@
+    """Validate a config option that should be a resolvable domain name"""
+    if not is_domain_name(value):
+        error(name, "{} is not a valid domain name".format(value))
+        return False
+
+    try:
+        socket.gethostbyname(value)
+        return True
+    except socket.gaierror:
+        error(name, "{} is not resolvable".format(value))
+        return False
+
+
+def ip_or_domain_name_with_port_validator(name, value):
+    """Validate a config option that should be a IP address or domain name,
+    followed by a port"""
+    match = re.match(r"^(.*):(\d+)$", value)
+    if not match:
+        error(name, "{} does not contain a port".format(value))
+        return False
+
+    stem = match.group(1)
+    port = match.group(2)
+
+    if int(port) >= 2**16:
+        error(name, "The port value ({}) is too large".format(port))
+        return False
+
+    if (stem[0] == '[') and (stem[-1] == ']'):
+        ip = stem[1:-1]
+        if ip_version(ip) == 6:
+            return True
+        else:
+            error(name, "{} is not a valid IPv6 address".format(ip))
+            return False
+
+    elif ip_version(stem) == 4:
+        return True
+
+    elif is_domain_name(stem):
+        return True
+
+    else:
+        error(name, ("{} is neither a domain name, "
+                     "IPv4 address, or bracketed IPv6 address").format(stem))
+        return False
+
+
+# Options that we wish to check.
+OPTIONS = [
+  Option('local_ip', True, ip_addr_validator),
+  Option('public_ip', True, ip_addr_validator),
+  Option('public_hostname', True, resolvable_domain_name_validator),
+  Option('etcd_cluster', True, ip_addr_list_validator),
+  Option('home_domain', True, domain_name_validator),
+  Option('sprout_hostname', True, ip_or_domain_name_validator),
+  Option('hs_hostname', True, ip_or_domain_name_with_port_validator),
+
+  Option('node_idx', False, integer_validator),
+  Option('ralf_hostname', False, ip_or_domain_name_with_port_validator),
+  Option('xdms_hostname', False, ip_or_domain_name_with_port_validator),
+]
+
+# Flag indicating whether all the config checks have passed. This affects the
+# script's exit code.
+all_ok = True
+
+# Check that each option is present (if mandatory) and correctly formatted (if
+# it has a particular format we wish to check).
+for option in OPTIONS:
+    value = os.environ.get(option.name)
+
+    if value:
+        # The option is present. If it has validator, run it now.
+        if option.validator:
+            if not option.validator(option.name, value):
+                # The validator is responsible for printing an error message.
+                all_ok = False
+
+    else:
+        # The option is not present, which is an error if it's mandatory.
+        if option.mandatory:
+            error(option.name, 'option is mandatory but not present')
+            all_ok = False
+
+#
+# More advanced checks (e.g. checking consistency between multiple options) can
+# be performed here.
+#
```

Comment:
Not at present. I've not implemented every check I could think of because that would take too long. Instead developers can add checks to this script as and when they seem useful. The same goes for these cross-option checks.